src/other/utils.jl

```diff
@@ -215,6 +324,89 @@ macro spawn_for_chunks(basesize, ex)
     return _spawn_for_chunks_helper(ex.args[1], ex.args[2], basesize)
 end
 
+"""
+    @spawn_or_async expr
+
+Equivalent to `Threads.@spawn` if [`DataFrames.ismultithreaded() === true`](@ref)
+and to `@async` otherwise.
+"""
+spawn_or_async
+
+"""
+    @spawn_or_run expr
+
+Equivalent to `Threads.@spawn` if [`DataFrames.ismultithreaded() === true`](@ref),
+otherwise simply runs `expr`.
+"""
+spawn_or_run
+
+if VERSION >= v"1.4"
+    macro spawn_or_async(expr)
+        letargs = Base._lift_one_interp!(expr)
+
+        thunk = esc(:(()->($expr)))
+        var = esc(Base.sync_varname)
+        quote
```

Comment:
Here and below there are some parts of code that are reported as not covered by tests.

Reply:
I suspect these are just bugs in coverage detection, as it doesn't make sense that the `if VERSION >= v"1.5"` path would not be tested but not the `else` branch would. Likewise, AFAICT line 385 is tested by test/utils.jl:193.